transaction/core/src/blockchain/quorum_set.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use alloc::{
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
+use core::{
+    hash::{Hash, Hasher},
+    str::FromStr,
+};
+use mc_common::{NodeID, ResponderId};
+use mc_crypto_digestible::Digestible;
+use mc_crypto_keys::Ed25519Public;
+use prost::{Message, Oneof};
+use serde::{Deserialize, Serialize};
+
+/// The quorum set defining the trusted set of peers.
+#[derive(Clone, Deserialize, Digestible, Message, Ord, PartialOrd, Serialize)]
+pub struct QuorumSet {
+    /// Threshold (how many members do we need to reach quorum).
+    #[prost(uint32, required, tag = 1)]
+    pub threshold: u32,
+
+    /// Members.
+    #[prost(message, repeated, tag = 2)]
+    pub members: Vec<QuorumSetMemberWrapper>,
```

Comment:
I am worried that having this definition differ from the one in the `.proto` file is going to cause us headaches in the future. Right now there is a 1:1 mapping between the two and afaik this is the only place in the codebase where the byte representation is not actually compatible.
This will be frustrating to debug if one is handed bytes encoded from the protobuf definition and tries to decode them using the Prost definition.

Reply:
The definition is indeed equivalent. There's a `quorum_set_round_trip` test (in `api/tests/prost.rs`) that verifies this.

I also inspected the code generated by both prost and protobuf, and it is using the same pattern, with the "wrapper" named `QuorumSetMember` and the Rust enum named `QuorumSetMember_oneof_member` (protobuf) or `quorum_set_member::Member` (prost puts it in a `mod`)